massa-protocol-worker/src/handlers/block_handler/retrieval.rs

```diff
@@ -463,6 +463,23 @@ impl RetrievalThread {
                 received: header.content.current_version,
             })
         } else {
+            // announced version == 0 -> No announce
+            if header.content.announced_version != 0 {
+                if header.content.announced_version <= version {
```

Comment:
```suggestion
                if header.content.announced_version < version {
```

Can't the announced version be the same as the current version? I'm pretty sure it can be as long as no new version has been pushed, so probably `<` instead of `<=`.

Reply:
I'm not a fan of special cases that need to be remembered "remember that announcing 0 means that there is no announcement" because it **will** cause errors. Indeed we should either announce the same version as the current version if we don't want to announce, or make the announcement an Option and set it to None

Reply:
It should not be allowed to have network versions like (1, 1). You can either have:
* (0, 1) -> announced a new network version: 1
* (1, 0) -> Version 1 is the current network version, 0 means nothing to announce
* (1, 2) -> Version 1 is the current network version, 2 is the announced network version

Reply:
I don't really mind about the solution we choose as long as its a logical behaviour

Reply:
I agree with the implementation and will resolve this, if any of the other participants want to reopen this discussion please feel free